core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -396,6 +396,8 @@ class SparkContext(config: SparkConf) extends Logging {
     if (!_conf.contains("spark.app.name")) {
       throw new SparkException("An application name must be set in your configuration")
     }
+    // This should be set as early as possible.
+    SparkContext.fillMissingMagicCommitterConfsIfNeeded(_conf)
```

Comment:
@dongjoon-hyun, should we maybe do this in `SparkSubmitArguments` with `sparkProperties` like `ignoreNonSparkProperties` handling? I think that;s as early as possible

Reply:
Sure, let me move this to there.